enterprise/frontend/src/embedding-sdk/hooks/private/use-load-static-question.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from "react";
+
+import { loadStaticQuestion } from "embedding-sdk/lib/load-static-question";
+import type { GenericErrorResponse } from "metabase/lib/errors";
+import { defer } from "metabase/lib/promise";
+import type Question from "metabase-lib/v1/Question";
+import type { Card, Dataset } from "metabase-types/api";
+
+type QuestionState = {
+  loading: boolean;
+  card: Card | null;
+  result: Dataset | null;
+  error: GenericErrorResponse | null;
+};
+
+export function useLoadStaticQuestion(
+  questionId: number | null,
+  parameterValues?: Record<string, string | number>,
+) {
+  const [questionState, setQuestionState] = useState<QuestionState>({
+    loading: false,
+    card: null,
+    result: null,
+    error: null,
+  });
+
+  const updateQuestion = (newQuestion: Question) =>
+    setQuestionState(state => ({
+      ...state,
+      card: newQuestion.card(),
+      loading: false,
+      error: null,
+    }));
+
+  useEffect(() => {
+    const cancelDeferred = defer();
+
+    async function loadCardData() {
+      setQuestionState(state => ({ ...state, loading: true }));
```

Comment:
I know this is moved from another file without changing the logic, but I wonder why we decided to set the `loading: true` even when there's no `questionId` passed, since technically you can't load a question without a question Id.

I'd expect we don't call this line until we have checked `if (!questionId)` first.